src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -325,6 +325,15 @@ private void WriteConstantOrParameter(CodeWriter writer, ReferenceOrConstant con
 
         private void WritePathSegment(CodeWriter writer, CodeWriterDeclaration uri, PathSegment segment, string? methodName = null)
         {
+            if (segment.Value.Type.IsFrameworkType &&
+                segment.Value.Type.FrameworkType == typeof(Uri))
+            {
+                writer.Append($"{uri}.Reset(");
+                WriteConstantOrParameter(writer, segment.Value, enumAsString: !segment.IsRaw);
+                writer.Line($");");
```

Comment:
Does this work for every situation? What if there are multiple Uri segments?

Reply:
Multiple Uri segment wouldn't make a lot of sense because all of them would have to be well formatted Uris.

Reply:
If we don't support relative URIs, that make sense.